docs/data/sdks/sdk-architecture.md

```diff
@@ -0,0 +1,340 @@
+
+You can configure the storage provider in the Client Configuration. Storage providers in different SDKs follows slightly different interfaces. Refer to the "Configuration Options" section of a specific next-gen SDK documentation for more details.
+
+```typescript
+// Next-gen Browser SDK example: Customize the storage
+
+class MyStorage<T> {
+  isEnabled(): Promise<boolean> {
+    // Whether the storage is enabled
+  }
+
+  get(key: string): Promise<T | undefined> {
+    // get the value associated with the given key in the storage
+  }
+
+  getRaw(key: string): Promise<string | undefined> {
+    // get the string associated with the given key in the storage
+  }
+
+  set(key: string, value: T): Promise<void> {
+    // set an item in the storage by key
+  }
+
+  remove(key: string): Promise<void> {
+    // remove an item in the storage by key
+  }
+
+  reset(): Promise<void> {
+    // clear the storage
+}
+
+amplitude.init('API_KEY', 'OPTIONAL_USER_ID', {
+  storageProvider: new MyStorage(),
+});
+```
+
+### Logger provider
+
+A logger provider configures the logger instance used by the Amplitude Client. It helps to collect debugging and error messages from the SDK in both development and production environments. Logger providers in different SDKs follows slightly different interfaces. Refer to the "Configuration Options" section of a specific next-gen SDK documentation for more details.
+
+```typescript
+// Next-gen Browser SDK example: Customize the logger 
+
+import * as amplitude from '@amplitude/analytics-browser';
+
+// Log levels defined:
+// None = 0,
+// Error = 1,
+// Warn = 2,
+// Verbose = 3,
+// Debug = 4,
+
+class MyLogger{
+  disable(): void {
+    console.log('Logger has been disabled');
+  }
+
+  enable(logLevel: amplitude.Types.LogLevel): void {
+    console.log('Logger has been enabled with level:', logLevel);
+  }
+
+  log(...args: any[]): void {
+    // Implementation for the log method
+    console.log(...args);
+  }
+
+  warn(...args: any[]): void {
+    // Implementation for the warn method
+    console.log(...args);
+  }
+
+  error(...args: any[]): void {
+    // Implementation for the error method
+    console.log(...args);
+  }
+
+  debug(...args: any[]): void {
+    // Implementation for the debug method
+    console.log(...args);
+  }
+}
+
+amplitude.init('API_KEY', 'OPTIONAL_USER_ID', {
+  loggerProvider: new MyLogger(),
+});
+```
+
+### Event 
+
+Events represent how users interact with your application. 
+
+`BaseEvent` represents a basic event with optional properties. You can also track other event properties in the field of `EventProperties`
+
+```typescript
+// Next-gen Browser SDK example: track events
+
+import * as amplitude from '@amplitude/analytics-browser';
+
+// Track a basic event
+amplitude.track('Button Clicked');
+
+// Track events with optional properties
+const eventProperties = {
+  buttonColor: 'primary',
+};
+amplitude.track('Button Clicked', eventProperties);
```

Comment:
Since this section is about Events vs track I think it would be good to focus on the event object and then mention that `track()` provides some convenience methods for sending Events by type, and type w/ properties, as they are a common case.

```
const myEvent: BaseEvent = {
  eventType: "My Event"
  eventProperties: {
    myEventProp: true,
  }
}

amplitude.track(myEvent);

// Send event by type only
amplitude.track('Button Clicked');

//Send event with type and properties
amplitude.track('Button Clicked', { buttonName: 'Success' });
```

Reply:
It makes sense to have both of them here. Changes applied